src/coreclr/jit/optimizer.cpp

```diff
@@ -6570,7 +6570,7 @@ PhaseStatus Compiler::optHoistLoopCode()
 #ifdef DEBUG
     // Test Data stuff..
     //
-    if (m_nodeTestData == nullptr)
+    if (m_nodeTestData != nullptr)
```

Comment:
Only reason the table was being created at all was because of this. We should probably just delete all this code...